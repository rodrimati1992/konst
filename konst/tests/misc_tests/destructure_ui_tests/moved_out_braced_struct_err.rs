struct Braced {
    bar: String,
    baz: String,
    qux: String,
}

const fn to_bar(foo: Braced) -> (String, String, String) {
    konst::destructure!{Braced{bar, baz, qux} = foo}

    std::mem::forget(foo);

    (bar, baz, qux)
}


fn main(){}