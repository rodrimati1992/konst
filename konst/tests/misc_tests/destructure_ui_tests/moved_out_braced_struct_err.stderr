error[E0382]: use of moved value: `foo`
  --> tests/misc_tests/destructure_ui_tests/moved_out_braced_struct_err.rs:10:22
   |
7  | const fn to_bar(foo: Braced) -> (String, String, String) {
   |                 --- move occurs because `foo` has type `Braced`, which does not implement the `Copy` trait
8  |     konst::destructure!{Braced{bar, baz, qux} = foo}
   |     ------------------------------------------------ value moved here
9  |
10 |     std::mem::forget(foo);
   |                      ^^^ value used here after move
